### ISSUE TITLE
Confusion about the different archetectures between code of VSSBlock and figure in paper of VSSBlock 

### ISSUE THREAD

**Issue by Allenem** [JCruan519/VM-UNet#56]
https://github.com/JCruan519/VM-UNet/blob/acbafd5ad5b275a0ddee5a449e2adeeb2243d6a4/models/vmunet/vmamba.py#L492

Hi, Thank you very much for your excellent work! 

I want to know why the `forward` function of `class VSSBlock` only includes 

**1** `layer norm`, `SS2D`, `drop` & `addition`; 

rather than what is ilustrated in paper Fig1 (b) : 

**2** `Layer Norm`, **3** `Linear layer`, **1** `DW-Conv`, **2** `Activation`, `SS2D`, `Addition` & **1** `Element-wise production`

**Comment by YunhengWu-IB** [comment]
I also would like to ask the same question!!
Do you know the reason now?

**Comment by Allenem** [comment]
> I also would like to ask the same question!! Do you know the reason now?

I understand the code now after reading the class `VSSBlock` & class `SS2D` together.

Both code and Fig1 (b) are same (only `drop` haven't been drawn in the figure). I add the comments to the fig1(b) here. The image below is easy to show the details. I hope you can understand the code easier after reading my comments 😄 :

![image](https://github.com/user-attachments/assets/eec03ba1-6939-438c-992e-8c442a2a33c8)

**Comment by YunhengWu-IB** [comment]
Thank you very much for your very nice explanation!!!

**Comment by z223668** [comment]
Your explanation and annotations are excellent. Thank you for your response

**Comment by LRuiRui517** [comment]
> > 我也想问同样的问题！！你现在知道原因了吗？
> 
> 我现在在阅读了课程和课程后理解了代码。`VSSBlock``SS2D`
> 
> 代码和图 1 （b） 是相同的（只是图中没有绘制）。我在此处将注释添加到图 1（b） 中。下图很容易显示细节。希望你看完我的评论😄后能更容易地理解代码：`drop`
> 
> ![image](https://github.com/user-attachments/assets/eec03ba1-6939-438c-992e-8c442a2a33c8)
So, can this "forward_core" be regarded as SSM?

**Comment by Allenem** [comment]
> So, can this "forward_core" be regarded as SSM?

Yes, it was implemneted by `mamba_ssm`. And you just need to feed some arguments (x, dt, A, B, C, D ...). You can see original repository for reference: 
https://github.com/state-spaces/mamba/blob/a6a1dae6efbf804c9944a0c2282b437deb4886d8/mamba_ssm/ops/selective_scan_interface.py#L106
